### PR TITLE
UI repo refactoring

### PR DESCRIPTION
**Pull request by Zequez**
Added a system of hooks to make the application state more modular. I'm still not completely satisfied, but I think it's a step in the right direction. Ideally we would have one hook for each contract, and store the libraries on the associated sub-app directory.

There were a lot of changes so there is a lot to review. It's going to be easier to just look at the new directory structure and check that the preview still looks the same. There should not be many visual changes (with some small-tweaks exceptions that I may have done along the way)

The goal is to make the repo easier to understand.

I still need to solve merge conflicts since main changed a bit (the Vercel fixes I think), so I'll do that and then we can merge.

### CODE REVIEW

**Comment by Zequez** [comment]
Deployment here https://ubiquity-dollar-staging-n1ulse6b6-ubiquity-vercel.vercel.app/ and here also https://github.com/ubiquity-vercel/ubiquity-dollar-staging/deployments/activity_log?environment=Preview

File: components/lib/contracts-shortcuts.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      return false;
+    }
+    const isAllowed2 = await contract.isApprovedForAll(signerAddress, spender);
+    console.log(`New ${logName} isAllowed: `, isAllowed2);
+  }
+
+  return true;
+}
+
+// const toEtherNum = (n: BigNumber) => +n.toString() / 1e18;
```
Comment (0x4007) [inline components/lib/contracts-shortcuts.ts:51]: @Zequez figure 1

File: components/monitor/TokenMonitor.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+};
+
+const TokenMonitor = (props: TokenMonitorProps) => {
+  return (
+    <Container className="col-span-2">
+      <Title text="Token monitor" />
+      <SubTitle text="Debt Coupon" />
+      <Address title="Debt Coupon Manager" address={props.debtCouponManagerAddress} />
+      <Address title="Debt Coupon" address={props.debtCouponAddress} />
+      <Balance title="Total Outstanding" unit="uDEBT" balance={props.totalOutstandingDebt} />
```
Comment (0x4007) [inline components/monitor/TokenMonitor.tsx:51]: We renamed the debts into credits!

File: components/price-stabilization/DebtCouponDeposit.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  return (
+    <>
+      <div className="flex flex-col">
+        <PositiveNumberInput value={inputVal} onChange={handleInput} placeholder="uAD Amount" />
+        <button className="btn-primary" onClick={handleBurn} disabled={!submitEnabled}>
+          Redeem uAD for uDEBT
+        </button>
+      </div>
+      {expectedDebtCoupon && inputVal && <p>expected uDEBT {formatEther(expectedDebtCoupon)}</p>}
```
Comment (0x4007) [inline components/price-stabilization/DebtCouponDeposit.tsx]: So again to clarify uDEBT -> uCR-NFT!

**Comment by Zequez** [comment]
I'll be closing this PR since this was already merged by the consolidated PR.

**Pull request closed without merging**